observIQ/bindplane-op-helm
# feat(auth): OIDC user authentication

## Description of Changes

This PR adds support for OIDC auth. OIDC auth is supported by BindPlane as of v1.84.0 (released 12/11/2024).

## Testing

If you want to run through this, see me for OAUTH client id and secret.

1. Deploy Minikube
```bash
minikube start
```

2. Create a license key secret
```bash
kubectl create secret generic bindplane \
  --from-literal=license=$BINDPLANE_LICENSE
```

3. Deploy postgres
```bash
kubectl apply \
  -f test/helper/postgres/postgres.yaml
```

Wait for Postgres w/ `k8s -n postgres`

4. Create values file. This config specifies OIDC and a pre release (v1.84.0) image. **Use your own oauth values or see me**
```yaml
config:
  sessions_secret: 4484766F-5016-4077-B8E0-0DE1D637854B
  server_url: http://localhost:3001
  licenseUseSecret: true

backend:
  type: postgres
  postgres:
    host: postgres.postgres.svc.cluster.local
    database: bindplane
    username: postgres
    password: password
    maxConnections: 20
    
eventbus:
  type: nats

replicas: 1

resources:
  requests:
    memory: 100Mi
    cpu: 100m
  limits:
    memory: 100Mi

nats:
  resources:
    requests:
      memory: 100Mi
      cpu: 100m
    limits:
      memory: 100Mi

auth:
  type: oidc
  oidc:
    issuer: "https://accounts.google.com"
    oauth2ClientID: "redacted"
    oauth2ClientSecret: "redacted"
    scopes: [
      "openid",
      "profile",
      "email"
    ]

image:
  name: bmedora/bindplane-ee-amd64
  tag: oidc.1
```

5. Deploy
```bash
helm upgrade \
  --install bindplane-ha \
  ./charts/bindplane \
  --values values.yaml
```

Wait for the pods to deploy.

```
NAME                                            READY   STATUS    RESTARTS   AGE
bindplane-ha-76c7f8f579-wlt9h                   1/1     Running   0          5m11s
bindplane-ha-jobs-7bbbb85559-qk4bv              1/1     Running   0          5m11s
bindplane-ha-nats-0                             1/1     Running   0          4m48s
bindplane-ha-nats-1                             1/1     Running   0          4m56s
bindplane-ha-nats-2                             1/1     Running   0          5m5s
bindplane-ha-prometheus-0                       1/1     Running   0          79m
bindplane-ha-transform-agent-67cdf57847-brqh6   1/1     Running   0          79m
```

7. Port forward
```bash
kubectl port-forward deploy/bindplane-ha 3001:3001
```

Navigate to http://localhost:3001 and login with your Gmail.

## **Please check that the PR fulfills these requirements**
- [ ] Changes to ports, services, or other networking have been tested with **istio**
